components/tdisplays3/display.py

```diff
@@ -9,38 +9,46 @@
 )
 from . import tdisplays3_ns
 
+DEPENDENCIES = ["esp32"]
+
+AUTO_LOAD = ["psram"]
+
+CONF_BACKLIGHT = "backlight"
+CONF_LOAD_FONTS = "load_fonts"
+CONF_LOAD_SMOOTH_FONTS = "load_smooth_fonts"
+
 TDISPLAYS3 = tdisplays3_ns.class_(
     "TDisplayS3", cg.PollingComponent, display.DisplayBuffer
 )
 
-def validate_tdisplays3(config):
-   return config
-
 CONFIG_SCHEMA = cv.All(
     display.FULL_DISPLAY_SCHEMA.extend(
         {
             cv.GenerateID(): cv.declare_id(TDISPLAYS3),
-            cv.Optional(CONF_HEIGHT): cv.int_,
-            cv.Optional(CONF_WIDTH): cv.int_,
+            cv.Optional(CONF_HEIGHT, default=320): cv.uint16_t,
+            cv.Optional(CONF_WIDTH, default=170): cv.uint16_t,
+            cv.Optional(CONF_BACKLIGHT, default=False): cv.boolean,
+            cv.Optional(CONF_LOAD_FONTS, default=False): cv.boolean,
+            cv.Optional(CONF_LOAD_SMOOTH_FONTS, default=False): cv.boolean,
         }
     ).extend(cv.polling_component_schema("5s")),
-    validate_tdisplays3,
 )
 
 async def to_code(config):
     # Add platformio build_flags for the correct TFT_eSPI settings for the T-Display-S3
     # This allows using current, unpatched versions of TFT_eSPI
-    cg.add_build_flag("-DUSER_SETUP_LOADED=1")
+    cg.add_build_flag("-DUSER_SETUP_LOADED")
     cg.add_build_flag("-DST7789_DRIVER")
     cg.add_build_flag("-DINIT_SEQUENCE_3")
     cg.add_build_flag("-DCGRAM_OFFSET")
     cg.add_build_flag("-DTFT_RGB_ORDER=TFT_RGB")
     cg.add_build_flag("-DTFT_INVERSION_ON")
     cg.add_build_flag("-DTFT_PARALLEL_8_BIT")
-    cg.add_build_flag("-DTFT_WIDTH=170")
-    cg.add_build_flag("-DTFT_HEIGHT=320")
-    cg.add_build_flag("-DTFT_DC=7")
+    cg.add_build_flag(f"-DTFT_WIDTH={config[CONF_WIDTH]}")
+    cg.add_build_flag(f"-DTFT_HEIGHT={config[CONF_HEIGHT]}")
     cg.add_build_flag("-DTFT_RST=5")
+    cg.add_build_flag("-DTFT_CS=6")
+    cg.add_build_flag("-DTFT_DC=7")
```

Comment:
made an issue to keep track of this

Reply:
This package is specific to the T-Display-S3. This board has the connection always on these pins. So I don't see a need to configure them ever.

Reply:
There have been comments on a different issue of people trying to use this repo for a different display (i.e. as a "generic TFT_eSPI display).
As I said, this is not strictly needed for this PR, but I would see this as a nice addition. Since Landon already created an issue for that, I would say let's leave it as it is for now.